### PR TITLE
Update ADR002, tutorial, and howto

### LINKED ISSUE #2093: Update the manual with `no-rows`
Once #2091 is merged, we have to update the manual with an explanation about `--features=imprecise.records`. Also, we need a short description of how to migrate from imprecise records to row records.

### PR DESCRIPTION
**Pull request by konnov**
Closes #2093.

This PR contains updates regarding the transition to Type Systems 1.2:

 - [x] ADR002
 - [x] The Snowcat tutorial
 - [x] The HOWTO on types

The manual contains two other documents that are using the old syntax for record types. They are postponed for #2099 and #2100.

### CODE REVIEW

File: docs/src/HOWTOs/howto-write-type-annotations.md [hunk 1]
```diff
@@ -492,6 +495,69 @@ Attacks(queens,i,j)
...
+   annotations over records and variants, whose shape is only
+   partially-defined.  For example, `{ foo: Int, bar: Bool, a }` defines a
+   record type that has at least two fields (that is, `foo` of type `Int` and
+   `bar` of type `Bool`), but may have more fields, which are captured with the
+   row variable `a`.
+
+### Case 1: plain records
+
+Many specifications are using plain records. For instance, they do not assign
+records of different domains to the same variable. Nor they mix records of
```
Comment (shonfeder) [inline docs/src/HOWTOs/howto-write-type-annotations.md:531]:
```suggestion
records of different domains to the same variable. Nor do they mix records of
```

File: docs/src/HOWTOs/howto-write-type-annotations.md [hunk 2]
```diff
@@ -492,6 +495,69 @@ Attacks(queens,i,j)
...
+different domains in the same set. Plenty of specifications fall into this
+class.
+
+For example, check [Recipe 2](#recipe2). In this recipe, the variable `chan` is
+always carrying a record of the domain `{ "val", "rdy", "ack" }`.
```
Comment (shonfeder) [inline docs/src/HOWTOs/howto-write-type-annotations.md:536]:
```suggestion
always carrying a record with the domain `{ "val", "rdy", "ack" }`.
```

File: docs/src/adr/002adr-types.md [hunk 3]
```diff
@@ -25,25 +25,40 @@ when writing TLA+ specifications. This document is filling this gap.
...
-        // warning: the following type will be removed in Type System 1.2
-      | '[' field ':' T ',' ...',' field ':' T ']'
+        // operators
       | '(' T ',' ...',' T ')' '=>' T
+        // constant types (uninterpreted types)
       | typeConst
+        // type variables
       | typeVar
+        // groups
       | '(' T ')'
```
Comment (shonfeder) [inline docs/src/adr/002adr-types.md:59]: What are "groups"?

File: docs/src/adr/002adr-types.md [hunk 4]
```diff
@@ -270,23 +278,26 @@ captures all interesting cases that occur in practice. Obviously, this type
...
+
+```tla
+{[type |-> "1a", bal |-> 1], [type |-> "2a", bal |-> 2, val |-> 3]}
+```
+
+This pattern often occurs in practice, e.g., see [Paxos][].  The type of that
+set will be `Set([type: Str, bal: Int, val: Int])`, which is probably not what
+you expected, but it is the best type we can actually compute without having
+algebraic datatypes in TLA+. It also reminds the user that one better tests the
+field `type` carefully.
```
Comment (shonfeder) [inline docs/src/adr/002adr-types.md:292]:
```suggestion
algebraic datatypes in TLA+. It also reminds the user that one must test the
field `type` carefully.
```

**Review by shonfeder — APPROVED** [review]
LGTM!

Just a few minor suggestions.

File: docs/src/adr/002adr-types.md (hunk 3 above)
Comment (konnov) [reply docs/src/adr/002adr-types.md:59]: well, these are just normal parentheses, which are normally use to change associativity. I will change it to something about associativity

**Pull request merged**